zkevm-circuits/src/witness/rw.rs

```diff
@@ -120,34 +125,60 @@ impl RwMap {
                     target_len, rows_len
                 );
             }
-            1
+            0
         }
     }
-    /// Prepad Rw::Start rows to target length
-    pub fn table_assignments_prepad(rows: &[Rw], target_len: usize) -> (Vec<Rw>, usize) {
-        // Remove Start rows as we will add them from scratch.
-        let rows: Vec<Rw> = rows
+    /// padding Rw::Start/Rw::Padding accordingly
+    pub fn table_assignments_padding(
+        rows: &[Rw],
+        target_len: usize,
+        is_first_row_padding: bool,
+    ) -> (Vec<Rw>, usize) {
+        // Remove Start/Padding rows as we will add them from scratch.
+        let rows_trimmed: Vec<Rw> = rows
             .iter()
-            .skip_while(|rw| matches!(rw, Rw::Start { .. }))
+            .filter(|rw| !(matches!(rw, Rw::Start { .. }) || matches!(rw, Rw::Padding { .. })))
             .cloned()
             .collect();
-        let padding_length = Self::padding_len(rows.len(), target_len);
-        let padding = (1..=padding_length).map(|rw_counter| Rw::Start { rw_counter });
-        (padding.chain(rows.into_iter()).collect(), padding_length)
+        let padding_length = {
+            let length = Self::padding_len(rows_trimmed.len(), target_len);
+            if is_first_row_padding {
+                length.saturating_sub(1)
+            } else {
+                length
+            }
+        };
+        let start_padding_rw_counter =
+            rows_trimmed.last().map(|row| row.rw_counter()).unwrap_or(1) + 1;
```

Comment:
There is an existing issue (might be no impact) since day one: rw_counter = 1 in rw_table is not unique. In fact, there will be another with non `Start` field tag also got `rw_counter = 1`. In state_circuit constraints, it seems we only emphasise rw_counter unique within same tag, but not across tag.

So the padding start from 1 is ok, which can be treat as special case

Reply:
I see, thanks for clarifying!